core/src/test/java/com/adobe/WorkspaceTest.java

```diff
@@ -11,8 +11,51 @@
  */
 package com.adobe;
 
+import com.adobe.util.Constants;
+import com.adobe.util.FileUtil;
+import java.io.IOException;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
 public class WorkspaceTest {
 
-  //TODO
+  @Rule
+  public ExpectedException expectedEx = ExpectedException.none();
+
+  private static final String TEST_PROPERTIES = "workspace.properties";
+  private static final String TEST_VALUE = "changeMe";
+
+  private static Workspace properties() throws IOException {
+    return Workspace.builder()
+        .properties(FileUtil.readPropertiesFromClassPath(TEST_PROPERTIES))
+        .build();
+  }
+
+  @Test
+  public void testProperties() throws IOException {
+    Workspace workspace = properties();
+    Assert.assertEquals(Constants.IMS_URL, workspace.getImsUrl());
+    Assert.assertEquals(TEST_VALUE, workspace.getApiKey());
+    Assert.assertEquals(TEST_VALUE, workspace.getClientSecret());
+    Assert.assertEquals(TEST_VALUE, workspace.getApiKey());
+    Assert.assertEquals(TEST_VALUE, workspace.getCredentialId());
+    Assert.assertEquals(TEST_VALUE, workspace.getClientSecret());
+    Assert.assertEquals(TEST_VALUE, workspace.getConsumerOrgId());
+    Assert.assertEquals(TEST_VALUE, workspace.getImsOrgId());
+    Assert.assertEquals(TEST_VALUE, workspace.getTechnicalAccountId());
+    Assert.assertEquals(TEST_VALUE, workspace.getMetascopes().iterator().next());
```

Comment:
nit: It will be better to use a different value for each field. This will make sure that the workspace POJO is managing/returning the correct value for each field.

Reply:
done